src/groupeddataframe/splitapplycombine.jl

```diff
@@ -1027,26 +1174,35 @@ function _combine_with_first(first::Union{NamedTuple, DataFrameRow, AbstractData
     elseif first isa DataFrameRow
         n = length(gd)
         eltys = [eltype(parent(first)[!, i]) for i in parentcols(index(first))]
-    else # NamedTuple giving a single row
+    elseif firstmulticol == Val(false) && first.x1 isa Union{AbstractArray{<:Any, 0}, Ref}
+        extrude_case = true
+        first = wrap_row(first.x1, firstmulticol)
+        n = length(gd)
+        eltys = (typeof(first.x1),)
```

Comment:
I'd use `first[1]` instead of `first.x1`, as the name is quite arbitrary.
```suggestion
    elseif firstmulticol == Val(false) && first[1] isa Union{AbstractArray{<:Any, 0}, Ref}
        extrude_case = true
        first = wrap_row(first[1], firstmulticol)
        n = length(gd)
        eltys = (typeof(first[1]),)
```